ci/scripts/cpp_build.sh

```diff
@@ -283,7 +283,10 @@ else
 fi
 
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
-  time meson install
+    time {
+      meson compile -j $[${n_jobs} + 1];
+      meson install;
+  }
```

Comment:
We need only compile time because install time will not be slow: 

```suggestion
  time meson compile -j $[${n_jobs} + 1]
  meson install
```